app/services/github.rb

```diff
@@ -1,13 +1,28 @@
 class Github
-  # Returns an array of hashes of all (public so far) unepwcmc repositories
-  def self.repos
-    response = HTTParty.get("https://api.github.com/orgs/unepwcmc/repos?client_id=#{Rails.application.secrets.github_key}&client_secret=#{Rails.application.secrets.github_secret}", headers: {"User-Agent" => "Labs"})
-    JSON.parse(response.body).map { |repo| OpenStruct.new(repo) }
+  def initialize
+    @repos
+  end
+
+  # Returns an array of hashes of all (public so far) unepwcmc repositories with the first element as a hash of link header info
+  # Pass in a page number for github pagination, defaults to 1
+  def get_all_repos page = 1
+    response = HTTParty.get("https://api.github.com/orgs/unepwcmc/repos?client_id=#{Rails.application.secrets.github_key}&client_secret=#{Rails.application.secrets.github_secret}&page=#{page}", headers: {"User-Agent" => "Labs"})
+    repos_array = JSON.parse(response.body).map { |repo| OpenStruct.new(repo) }
+    @repos = repos_array.unshift(parse_link_headers(response.headers['link']))
```

Comment:
This is a really curious way of returning more than one type of result from a method. If you really need to do that, it is usually more intuitive to return a hash that would have separate keys for repos / link headers. However, that would still not be particularly elegant. A more OO API could allow you to do something like this:

ghr = GithubRepoList.new(1)
ghr.repos
ghr.next_page
ghr.previous_page

That would be more in line with single responsibility principle and easier to use in the views.